Sources/Lumina/Camera/Extensions/SessionConfigurationExtension.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  SessionConfigurationExtension.swift
+//  Lumina
+//
+//  Created by David Okun on 11/20/17.
+//  Copyright © 2017 David Okun. All rights reserved.
+//
+
+import Foundation
+import AVFoundation
+
+extension LuminaCamera {
+  func requestVideoPermissions() {
+    self.sessionQueue.suspend()
+    AVCaptureDevice.requestAccess(for: .video) { success in
+      if success {
+        LuminaLogger.notice(message: "successfully enabled video permissions")
+        self.sessionQueue.resume()
+        self.delegate?.cameraSetupCompleted(camera: self, result: .requiresUpdate)
+      } else {
+        LuminaLogger.warning(message: "video permissions were not allowed - video feed will not show")
+        self.delegate?.cameraSetupCompleted(camera: self, result: .videoPermissionDenied)
+      }
+    }
+  }
+  
+  func requestAudioPermissions() {
+    self.sessionQueue.suspend()
+    AVCaptureDevice.requestAccess(for: AVMediaType.audio) { success in
+      if success {
+        LuminaLogger.notice(message: "successfully enabled audio permissions")
+        self.sessionQueue.resume()
+        self.delegate?.cameraSetupCompleted(camera: self, result: .requiresUpdate)
+      } else {
+        LuminaLogger.warning(message: "audio permissions were not allowed - audio feed will not be present")
+        self.delegate?.cameraSetupCompleted(camera: self, result: .audioPermissionDenied)
+      }
+    }
+  }
+  
+  func updateOutputVideoOrientation(_ orientation: AVCaptureVideoOrientation) {
+    self.videoBufferQueue.async {
+      for output in self.session.outputs {
+        guard let connection = output.connection(with: AVMediaType.video) else {
+          continue
+        }
+        if connection.isVideoOrientationSupported {
+          connection.videoOrientation = orientation
+        }
+      }
+    }
+  }
+  
+  func restartVideo() {
+    LuminaLogger.notice(message: "restarting video feed")
+    if self.session.isRunning {
+      self.stop()
+      updateVideo { result in
+        if result == .videoSuccess {
+          self.start()
+        } else {
+          self.delegate?.cameraSetupCompleted(camera: self, result: result)
+        }
+      }
+    }
+  }
+  
+  func updateAudio(_ completion: @escaping (_ result: CameraSetupResult) -> Void) {
+    self.sessionQueue.async {
+      self.purgeAudioDevices()
+      switch AVCaptureDevice.authorizationStatus(for: AVMediaType.audio) {
+        case .authorized:
+          guard let audioInput = self.getNewAudioInputDevice() else {
+            return completion(CameraSetupResult.invalidAudioInput)
+          }
+          guard self.session.canAddInput(audioInput) else {
+            return completion(CameraSetupResult.invalidAudioInput)
+          }
+          self.audioInput = audioInput
+          self.session.addInput(audioInput)
+          return completion(CameraSetupResult.audioSuccess)
+        case .denied:
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/5e323e0c2296f8f5c5b041c942e35a2720d9c81e/Sources/Lumina/Camera/Extensions/SessionConfigurationExtension.swift#L82"></span>Case statements should vertically align with their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>SessionConfigurationExtension.swift:82</code></td>
    </tr>
  </tbody>
</table>